### PR TITLE
Development

### PR DESCRIPTION
**Pull request by rvvergara**
Version 1 of our Project: 
- code cleanup still needed
- frontend view not yet refined
- only basic functionalities were built:
 > showing all books in the library
 > adding a book to the library (ergo, creating an instance of Book class [we used class because we like it]) and showing it on display
> updating the "Read?" attribute of each book through a Toggle button
> removing a book from the library and from the display

=> live demo: https://rawcdn.githack.com/rvvergara/library/f9ce007f9d37b464d6b6fb1780c060a0fedf2f79/index.html

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  // 1. Instantiate a new book
+  let book = create(...arguments);
+  // 2. Push this new book to myLibrary
+  myLibrary.push(book);
+  return myLibrary;
+}
+
+function removeBook(booksArr, param) {
+  // Find book from array using param -> identification of a particular book and then find its index in the library
+  index = param;
```
Comment (heyapricot) [inline assets/js/app.js]: `index` is not defined

File: assets/js/app.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  })
+}
+
+
+// Function to show form
+function showForm() {
+  document.getElementById("addBookForm").removeAttribute("class");
+}
+
+window.addEventListener("load", e => render(myLibrary));
```
Comment (heyapricot) [inline assets/js/app.js:6]: I'd rather use an [IIFE](https://stackoverflow.com/questions/22537211/should-i-use-iife-or-window-onload-to-initialize)

File: assets/js/app.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+function removeBook(booksArr, param) {
+  // Find book from array using param -> identification of a particular book and then find its index in the library
+  index = param;
+  booksArr.splice(index, 1);
+  return booksArr;
+}
+
+
+// Function to create/instantiate a new book
+function create(title, author, pages, read) {
```
Comment (heyapricot) [inline assets/js/app.js]: This function is not necessary

**Review by heyapricot — CHANGES_REQUESTED** [review]
# You are getting closer 🚂
Hello! 👋🏼 I can tell you have a good idea of what you want to achieve and you are getting close. However, I have some requests.

1. Add styling. Quoting from the project page 
>You can display them in some sort of table, or each on their own “card”.

2. The app.js file is hard to read. Rearrange the file so that the function definitions come first, then variables and then the logic, all the function calls.

After doing those changes please submit a new code review request. Thanks!

**Comment by heyapricot** [comment]
# Project completed ☑️

Ready to move on. No need from my side for further review.

File: assets/js/book.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Book Constructor
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+  Book.count++;
+  // Generate randomized unique ID for every new book
+  this.id = Math.round(Math.random() * 1000000000);
```
Comment (heyapricot) [inline assets/js/book.js:9]: This is a Hack and if by chance we end up with two elements having the same id there's going to be trouble. What happened to the static variable?

**Review by heyapricot — APPROVED** [review]

File: assets/js/book.js (hunk 4 above)
Comment (rvvergara) [reply assets/js/book.js:9]: Yup we just wanted to minimize any access to the Book.count. . But yeah you're right :) We just couldn't figure out how to make Book.count private

**Comment by rvvergara** [comment]
> # Project completed ballot_box_with_check
> 
> Ready to move on. No need from my side for further review.

Muchos gracias Alvaro! :)

**Pull request merged**